lib/ts/ink_memory.h

```diff
@@ -133,6 +133,16 @@ static inline size_t __attribute__((const)) ats_pagesize(void)
 char *_xstrdup(const char *str, int length, const char *path);
 
 #define ats_strdup(p) _xstrdup((p), -1, nullptr)
+
+// this is to help with migration to a std::string issue with older code that
+// expects char* being copied. As more code moves to std::string, this can be
+// removed to avoid these extra copies.
+inline char *
+ats_stringdup(std::string const &p)
+{
+  return _xstrdup((p.empty() ? nullptr : p.c_str()), -1, nullptr);
```

Comment:
This seems slightly inefficient ... why not

```
return p.empty() ? nullptr : _xstrdup(p.c_str(), ...
```

Also, rather than -1, pass in the length from the std::string (why call strlen on the string).